tree/dataframe/test/dataframe_concurrency.cxx

```diff
@@ -76,4 +77,72 @@ TEST(RDFSimpleTests, ThrowOnPoolSizeMismatch)
       }
    }
 }
+
+
+void SimpleParallelRDFs()
+{
+   // Run the RDF construction and the event loop in parallel
+   auto func = [] {
+         ROOT::RDataFrame df(10);
+         return df.Define("x", "rdfentry_").Mean("x").GetValue();
+      };
+
+   ROOT::TThreadExecutor pool;
+   auto res = pool.Map(func, 8);
+
+   const auto ref = func();
+
+   for(auto& r : res) {
+      EXPECT_EQ(r, ref);
+   }
+}
+
+TEST(RDFConcurrency, SimpleParallelRDFsEnableThreadSafety)
+{
+   ROOT::EnableThreadSafety();
+   SimpleParallelRDFs();
+}
+
+TEST(RDFConcurrency, SimpleParallelRDFsEnableImplicitMT)
+{
+   ROOT::EnableImplicitMT();
+   SimpleParallelRDFs();
+   ROOT::DisableImplicitMT();
+}
+
+void SimpleParallelRDFLoops()
+{
+   // Run only the event loops in parallel
+   auto create_df = [] {
+         ROOT::RDataFrame df(10);
+         return df.Define("x", "rdfentry_").Mean("x");
+      };
+
+   std::vector<ROOT::RDF::RResultPtr<double>> vals(10);
+   for (int i = 0; i < (int)vals.size(); i++)
+       vals[i] = create_df();
+
+   ROOT::TThreadExecutor pool;
+   auto func = [](ROOT::RDF::RResultPtr<double> rptr){ return rptr.GetValue(); };
+   auto res = pool.Map(func, vals);
+
+   auto ref = func(create_df());
+
+   for(auto& r : res) {
+      EXPECT_EQ(r, ref);
+   }
+}
+
+TEST(RDFConcurrency, SimpleParallelRDFLoopsEnableThreadSafety)
+{
+   ROOT::EnableThreadSafety();
+   SimpleParallelRDFLoops();
+}
+
+TEST(RDFConcurrency, SimpleParallelRDFLoopsEnableImplicitMT)
+{
+   ROOT::EnableImplicitMT();
+   SimpleParallelRDFLoops();
+   ROOT::DisableImplicitMT();
+}
 #endif
```

Comment:
we definitely need to also test concurrent `Snapshot`s (I don't expect surprises, but it's good to have a test) and `SaveGraph` (we saw above it has these static counter thingies)